From f6ae03bf838f5ce281999aeb5fe3d4ab83662bf0 Mon Sep 17 00:00:00 2001
From: Chris Burroughs <chris.burroughs@joyent.com>
Date: Fri, 1 Mar 2019 15:36:19 -0500
Subject: [PATCH] TRITON-1275 imgapi runtests colorization induces spurious
 failures Reviewed by: Trent Mick <trentm@gmail.com> Reviewed by: Todd
 Whiteman <todd.whiteman@joyent.com> Approved by: Trent Mick
 <trentm@gmail.com> Approved by: Todd Whiteman <todd.whiteman@joyent.com>

---
 test/runtests | 34 ++--------------------------------
 1 file changed, 2 insertions(+), 32 deletions(-)

diff --git a/test/runtests b/test/runtests
index 7ec6ba5..a90ed7c 100755
--- a/test/runtests
+++ b/test/runtests
@@ -305,41 +305,11 @@ if [[ -n "$test_files" ]]; then
     if [[ $opt_reporter == "tap" ]]; then
         PATH=$NODE_INSTALL/bin:$PATH $NODEUNIT --reporter $opt_reporter $test_files \
             | tee $OUTPUT_DIR/imgapi.tap
+        retval=$?
     else
         PATH=$NODE_INSTALL/bin:$PATH $NODEUNIT --reporter $opt_reporter $test_files
         retval=$?
     fi
 fi
 
-
-if [[ $opt_reporter == "tap" ]]; then
-    echo ""
-    echo "# test output:"
-    ls $OUTPUT_DIR/*.tap
-
-    # Colored summary of results (borrowed from smartos-live.git/src/vm/run-tests).
-    echo ""
-    echo "# test results:"
-
-    end_time=$(date +%s)
-    elapsed=$((${end_time} - ${start_time}))
-
-    tests=$(grep "^# tests [0-9]" $OUTPUT_DIR/*.tap | cut -d ' ' -f3 | xargs | tr ' ' '+' | bc)
-    passed=$(grep "^# pass  [0-9]" $OUTPUT_DIR/*.tap | tr -s ' ' | cut -d ' ' -f3 | xargs | tr ' ' '+' | bc)
-    [[ -z ${tests} ]] && tests=0
-    [[ -z ${passed} ]] && passed=0
-    fail=$((${tests} - ${passed}))
-
-    echo "# Completed in ${elapsed} seconds."
-    echo -e "# \033[32mPASS: ${passed} / ${tests}\033[39m"
-    if [[ ${fail} -gt 0 ]]; then
-        echo -e "# \033[31mFAIL: ${fail} / ${tests}\033[39m"
-    fi
-    echo ""
-
-    if [[ ${tests} != ${passed} ]]; then
-        exit 1
-    fi
-else
-    exit $retval
-fi
+exit ${retval}
-- 
2.21.0

